test/common/runtime/runtime_impl_test.cc

```diff
@@ -1102,17 +1102,19 @@ TEST_F(RtdsLoaderImplTest, OnConfigUpdateSuccess) {
     layer:
       foo: bar
       baz: meh
+      toggle: true
   )EOF");
   EXPECT_CALL(rtds_init_callback_, Call());
   doOnConfigUpdateVerifyNoThrow(runtime);
 
   EXPECT_EQ("bar", loader_->snapshot().get("foo").value().get());
   EXPECT_EQ("yar", loader_->snapshot().get("bar").value().get());
   EXPECT_EQ("meh", loader_->snapshot().get("baz").value().get());
+  EXPECT_EQ("true", loader_->snapshot().get("toggle").value().get());
```

Comment:
just to confirm, this check fails prior to your change?

Reply:
Yes. This is a newly added check.

Without the source code change, it will fail, since snapshot().get("toggle").value().get() will return `"1"` instead of `"true"`.

Reply:
Also updated a comment so it is more clear why `absl::StrCat` is not used for boolean case